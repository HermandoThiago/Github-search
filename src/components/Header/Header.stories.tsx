import { Meta, Story } from "@storybook/react";
import { IPropsHeader, Header } from "./Header";

export default {
  title: "Components/Header",
  component: Header,
  args: {
    children: "Github Search",
  },
} as Meta;

export const Default: Story<IPropsHeader> = () => (
  <Header>Github Search</Header>
);
