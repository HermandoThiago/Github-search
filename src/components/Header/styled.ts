import styled from "styled-components";

export const StyledHeader = styled.header`
    width: 100%;
    padding: 20px 0;
    color: #fff;
    text-align: center;
    background-color: #2A2550;
    font-size: 1.5rem;
    font-weight: bold;
`;
