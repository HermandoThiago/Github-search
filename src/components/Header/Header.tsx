import { StyledHeader } from "./styled"

interface IPropsHeader {
    children: string | React.ReactNode;
}

export default function Header({ children }: IPropsHeader){
    return (
        <StyledHeader>
            {children}
        </StyledHeader>
    )
}