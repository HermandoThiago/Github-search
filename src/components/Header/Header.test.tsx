import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Header } from "./Header";

describe("Testing Header", () => {
  it("Render correct test", async () => {
    render(<Header>Github Search</Header>);

    await screen.findByText(/Github Search/i);
  });

  it("should match snapshot", () => {
    const component = renderer.create(<Header>Github Search</Header>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
