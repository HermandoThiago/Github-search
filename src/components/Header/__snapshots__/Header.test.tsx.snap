// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header should match snapshot 1`] = `
<header
  className="sc-bczRLJ eVkjkM"
>
  Github Search
</header>
`;
